src/os/command/feature/featurecolorcmd.js

```diff
@@ -41,6 +39,8 @@ os.command.FeatureColor = function(layerId, featureId, color, opt_oldColor, opt_
       break;
   }
 
+  os.command.FeatureColor.base(this, 'constructor', layerId, featureId, color, opt_oldColor);
```

Comment:
The value for 'this.changeMode' needed to be set before the parent constructor was run. If it is not set and no 'opt_oldValue' is set, the parent constructor calls this.getOldValue() to determine the previous value. Without that change, any undo action was getting the default opacity instead of properly getting the fill's previous opacity.

That said, the constructor should have been moved to just below setting 'this.changeMode' instead of where I previously moved it. This function has been updated.